library/ssl_tls13_server.c

```diff
@@ -1633,6 +1633,49 @@ static int ssl_tls13_parse_client_hello(mbedtls_ssl_context *ssl,
     return hrr_required ? SSL_CLIENT_HELLO_HRR_REQUIRED : SSL_CLIENT_HELLO_OK;
 }
 
+#if defined(MBEDTLS_SSL_EARLY_DATA)
+static void ssl_tls13_update_early_data_state(mbedtls_ssl_context *ssl)
+{
+    mbedtls_ssl_handshake_params *handshake = ssl->handshake;
+
+    if ((handshake->received_extensions &
+         MBEDTLS_SSL_EXT_MASK(EARLY_DATA)) == 0) {
+        MBEDTLS_SSL_DEBUG_MSG(
+            1, ("EarlyData: early data extension is not received."));
+        return;
+    }
+
+    handshake->early_data_state = MBEDTLS_SSL_EARLY_DATA_SRV_STATE_REJECTED;
+
+    if (ssl->conf->early_data_enabled == MBEDTLS_SSL_EARLY_DATA_DISABLED) {
+        MBEDTLS_SSL_DEBUG_MSG(
+            1, ("EarlyData: rejected. configured disabled."));
+        return;
+    }
+
+    if (ssl->conf->max_early_data_size > MBEDTLS_SSL_MAX_EARLY_DATA_SIZE) {
```

Comment:
I do not understand why you do this. In mbedtls_ssl_read_early_data() we plan to store early data into buffers provided by the application thus I do not think our TLS 1.3 implementation should impose a compile time limit: MBEDTLS_SSL_MAX_EARLY_DATA_SIZE here.

Reply:
I think your are right.  will remove that